src/Command/MakeSerializerEncoderCommand.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Command;
+
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\Str;
+use Symfony\Bundle\MakerBundle\Validator;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\Question;
+use Symfony\Component\Serializer\Serializer;
+
+/**
+ * @author Piotr Grabski-Gradzinski <piotr.gradzinski@gmail.com>
+ */
+final class MakeSerializerEncoderCommand extends AbstractCommand
+{
+    protected static $defaultName = 'make:serializer:encoder';
+
+    protected function configure()
+    {
+        $this
+            ->setDescription('Creates a new custom encoder class')
+            ->addArgument('name', InputArgument::OPTIONAL, 'Choose a class name for your encoder (e.g. <fg=yellow>YamlEncoder</>).')
+            ->addArgument('format', InputArgument::OPTIONAL, 'Pick your format name (e.g. <fg=yellow>yaml</>)')
+            ->setHelp(file_get_contents(__DIR__.'/../Resources/help/MakeSerializerEncoder.txt'))
+        ;
+    }
+
+    protected function getParameters(): array
+    {
+        $encoderClassName = Str::asClassName($this->input->getArgument('name'), 'Encoder');
+        Validator::validateClassName($encoderClassName);
+        $format = $this->input->getArgument('format');
+
+        return [
+            'encoder_class_name' => $encoderClassName,
+            'format' => $format,
+        ];
+    }
+
+    protected function getFiles(array $params): array
+    {
+        return [
+            __DIR__.'/../Resources/skeleton/serializer/Encoder.php.txt' => 'src/Serializer/'.$params['encoder_class_name'].'.php'
+        ];
+    }
+
+    protected function writeNextStepsMessage(array $params, ConsoleStyle $io)
+    {
+        $io->text([
+            'Next: Open your new encoder class and start customizing it.',
```

Comment:
This is not a question about your Pull Request but a general question: are these "next steps" instructions really useful?

* This doesn't provide much: *"Next: Open your new encoder class and start customizing it."*
* And this is more suited for a "demo app" or "newcomer learning resource": *"Find the documentation at ..."*

I propose to remove all these obvious "next steps" and only keep the ones really needed.

Reply:
I think you may be right. But I still want a clear “success” message at the bottom. Maybe there’s a default, generic success message and we only override it when there really are additional steps

Reply:
The exact *next* information is not really helpful in this case (just followed the convention from other commands). 

Just to point that when you generate new command over SensioGeneratorBundle after confirmation you'll receive

```
You are going to generate a foo-bar command inside AppBundle bundle.
Do you confirm generation [yes]? 
  created ./src/AppBundle/Command/
  created ./src/AppBundle/Command/FooBarCommand.php
Generated the foo-bar command in AppBundle

                                         
  Everything is OK! Now get to work :).  
```

Not sure if and how close to this maker should be.

If agreed I can refactor existing commands for that.

Reply:
OK ... let's discuss about this in a separate PR and, if needed, let's make the change at once for all commands.